core/tests/puppeteer-acceptance-tests/utilities/common/puppeteer-utils.ts

```
@@ -82,7 +86,7 @@ export class BaseUser {
 
     const headless = process.env.HEADLESS === 'true';
     const mobile = process.env.MOBILE === 'true';
-    const specName = process.env.SPEC_NAME;
+    this.specName = process.env.SPEC_NAME ?? '';
```

Why is this change needed?

---

Because we are using `specName` in new `setupVideoRecording` method.

---

No, I meant the `?? ''` part (sorry for not being clear).

---

I declared `specName` attribute as string. So, we are required to have `?? ''`, as `process.env.SPEC_NAME` can be `null` (different type than string).

---

OK, maybe worth explaining in the code then.

Though perhaps better to fail noisily if it's null?

---

It reverted back to original while moving implementation of screen recording from its own method to openBrowser method.